lib/value/src/kind/comparison.rs

```diff
@@ -86,7 +92,6 @@ impl Kind {
     pub const fn is_float(&self) -> bool {
         self.bytes.is_none()
             && self.integer.is_none()
-            && self.float.is_some()
             && self.boolean.is_none()
```

Comment:
Is `&& self.float.is_some()` implied now? Is it because all types are `never`?

Reply:
a `never` can be considered a float. So the value of `float` is this case doesn't actually matter. What's important is that it can't be any _other_ type.

Reply:
"it can never produce a type that is not a float" is probably a better way to describe it.